cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -77,6 +77,10 @@ spec:
         k8s-app: kube-dns
     spec:
       priorityClassName: system-cluster-critical
+      securityContext:
+        sysctls:
+        - name: net.ipv4.ip_unprivileged_port_start
```

Comment:
I like the change but the whole point of this is that the containers can now be unprivileged? So can we set change the UID and GID we run this as, using runAsUser and runAsGroup?

Reply:
/cc @chrisohaver @rajansandeep

do you think we can switch the kubeadm coredns pods to non-0 GID/UID?

Reply:
Yes, we can.

Reply:
yes, i have assumed so.
pinged the coredns maintainers to see if they are confident about the `net.ipv4.ip_unprivileged_port_start` change and this PR in general.

Reply:
I ... need to research this to understand it better before I can weigh in.

Reply:
I don't foresee any issues with this.

Reply:
@vinayakankugoyal maybe we can use `sysctl` and non-root user for coredns if RootlessControlPlane is enabled.